src/textual/renderables/opacity.py

```diff
@@ -0,0 +1,90 @@
+import functools
+
+from rich.color import Color
+from rich.console import ConsoleOptions, Console, RenderResult, RenderableType
+from rich.segment import Segment
+from rich.style import Style
+
+from textual.renderables.utilities import blend_colors
+
+
+class Opacity:
+    """Wrap a renderable to blend foreground color into the background color.
+
+    Args:
+        renderable (RenderableType): The RenderableType to manipulate.
+        value (float): The opacity as a float. A value of 1.0 means text is fully visible.
+    """
+
+    def __init__(self, renderable: RenderableType, value: float = 1.0) -> None:
+        self.renderable = renderable
+        self.value = value
+
+    def __rich_console__(
+        self, console: Console, options: ConsoleOptions
+    ) -> RenderResult:
+        segments = console.render(self.renderable, options)
+        opacity = self.value
+        for segment in segments:
+            style = segment.style
+            if not style:
+                yield segment
+                continue
+            fg, bg = style.color, style.bgcolor
```

Comment:
I suspect this is slower that doing the assignment on two lines, since you would need to construct a tuple.

Although, if you ran a `timeit` and this proved not the case, I would happily withdraw my objection!